Lambda-scenarios/datasync_testing.py.txt

```diff
@@ -0,0 +1,164 @@
+
+#
+# Lambda function used to write inbound IoT sensor data to RDS PostgeSQL database
+#
+import sys
+import os
+import json
+import psycopg2
+import boto3
+import base64
+from botocore.exceptions import ClientError
+import logging
+from datetime import datetime
+
+secret_name = "arn:aws:secretsmanager:eu-west-2:063935053328:secret:bip_db_pass-b2Yqp4"
+# region_name = os.environ['AWS_REGION']
+
+# Enable the logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s", datefmt='%Y-%m-%d %H:%M:%S %Z')
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+sampleAPIJSON={"sourceLocation": "/ghdevhome/binfs/210303_A01021_0222_BHY2V2DSXY/",
+               "destinationLocation": "bip-analysis-bucket",
+               "taskname": "GH_BIP_TASK"}
+
+# Connect to AWS boto3 Client
+def aws_connect_client(service):
+    try:
+        # Gaining API session
+        #session = boto3.Session(aws_access_key_id=ACCESS_KEY, aws_secret_access_key=SECRET_KEY)
+        session = boto3.Session()
+        my_session = boto3.session.Session()
+        REGION = my_session.region_name
+        # Connect the resource
+        conn_client = session.client(service, REGION)
+    except Exception as e:
+        logger.error('Could not connect to region: %s and resources: %s , Exception: %s\n' % (REGION, service, e))
+        conn_client = None
+    return conn_client, REGION
+
+def getAccountID():
+    conn,_ = aws_connect_client("sts")
+    try:
+        account_id = conn.get_caller_identity()["Account"]
+    except Exception as err:
+        logger.error(f"Unable to get Account ID. Exception: {err}")
+        sys.exit(1)
+    return account_id
+
+def getDBConnection():
+    # conn = psycopg2.connect(host='bip-db.cluster-c5jsjiocmimr.eu-west-2.rds.amazonaws.com', port=5432, dbname='bipanalysisdb', user='bipadmin', password=secret['bip_db_pass'])
+
+    conn = psycopg2.connect(host='localhost', port=5432, dbname='test')

```

Comment:
Why db host as local host ? check the table creation code and see how db connection parameters are defined